src/test/kotlin/ecommerce/controller/ProductControllerTest.kt

```diff
@@ -0,0 +1,155 @@
+            .then()
+            .extract()
+            .response()
+        RestAssured
+            .given().log().all().body(product2)
+            .contentType(ContentType.JSON)
+            .`when`()
+            .request("POST", "/api/products")
+            .then()
+            .extract()
+            .response()
+        RestAssured
+            .given().log().all().body(product3)
+            .contentType(ContentType.JSON)
+            .`when`()
+            .request("POST", "/api/products")
+            .then()
+            .extract()
+            .response()
+    }
+
+    private fun createQuery(): String {
+        return """
+            create table PRODUCTS
+            (
+                ID       int              not null AUTO_INCREMENT,
+                NAME     varchar(100)     not null,
+                PRICE    double not null,
+                IMAGE_URL varchar(500),
+                PRIMARY KEY (ID)
+            )
+            """.trimIndent()
+    }
+
+    @Test
+    fun addProduct() {
+        val response =
+            RestAssured
+                .given().log().all().body("""{ "name": "Screen", "price": 190.0, "imageUrl": "ciao" }""")
+                .contentType(ContentType.JSON)
+                .`when`()
+                .request("POST", "/api/products")
+                .then()
+                .extract()
+                .response()
+
+        val newProduct =
+            RestAssured
+                .given().log().all()
+                .contentType(ContentType.JSON)
+                .`when`()
+                .request("GET", "/api/products/4")
+                .then()
+                .extract()
+                .response()
+
+        assertThat(response.statusCode).isEqualTo(HttpStatus.CREATED.value())
+        assertThat(newProduct.`as`(Product::class.java).name).isEqualTo("Screen")
+    }
+
+    @Test
+    fun getProducts() {
+        val response =
+            RestAssured
+                .given().log().all()
+                .`when`()
+                .request("GET", "/products")
+                .then()
+                .extract()
+                .response()
+
+        assertThat(response.statusCode).isEqualTo(HttpStatus.OK.value())
+    }
+
+    @Test
+    fun getProduct() {
+        val response =
+            RestAssured
+                .given().log().all()
+                .`when`()
+                .request("GET", "/api/products/1")
+                .then()
+                .extract()
+                .response()
+
+        assertThat(response.statusCode).isEqualTo(HttpStatus.OK.value())
+        assertThat(response.`as`(ProductResponse::class.java).id).isEqualTo(1L)
+        assertThat(response.`as`(ProductResponse::class.java).name).isEqualTo("cafe")
+        assertThat(response.`as`(ProductResponse::class.java).price).isEqualTo(39.00)
+        assertThat(response.`as`(ProductResponse::class.java).imageUrl).isEqualTo("www.test")
+    }
+
+    @Test
+    fun deleteProduct() {
```

Comment:
<img width="1552" height="987" alt="image" src="https://github.com/user-attachments/assets/dfbab68d-c1a6-4af5-8507-5232de164ce9" />

Could you take a look at this test?